server/lib/chat/chat-service.ts

```diff
@@ -490,33 +497,48 @@ export default class ChatService {
     })
   }
 
-  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<ChannelInfo> {
+  async getChannelInfo(channelId: SbChannelId, userId: SbUserId): Promise<GetChannelInfoResponse> {
     const channelInfo = await getChannelInfo(channelId)
+    // TODO(2Pac): Get this from DB instead.
+    const isUserInChannel = this.state.users.get(userId)?.has(channelId)
 
     if (!channelInfo) {
       throw new ChatServiceError(ChatServiceErrorCode.ChannelNotFound, 'Channel not found')
     }
 
     let userCount
-    if (!channelInfo.private || this.state.users.get(userId)?.has(channelId)) {
+    if (!channelInfo.private || isUserInChannel) {
```

Comment:
nit: I think all of this would be more clear as a ternary within the return now

If userCount is *also* private (like description) it might be more worthwhile to make the 3rd struct we had talked about, I don't see how you're going to easily keep from overwriting the userCount in other requests otherwise so private channels will have bugs

Reply:
Yeah, imo you should move to a very simple `ChannelInfo` that is stuff that is always present (name, mostly) and is what is given as dat for channel mentions, another struct for things that are visible only for public + joined channels (ProtectedChannelData? SensitiveChannelData? I dunno, something like that). Might pay to write a db query that returns the small ChannelInfo directly as well, since that's what we'd be querying in bulk and it can be returned using only indexes, but it's a bit of a premature optimization maybe